modules/mesh-task/templates/consul_client_command.tpl

```diff
@@ -2,36 +2,23 @@ cp /bin/consul /bin/consul-inject/consul
 
 ECS_IPV4=$(curl -s $ECS_CONTAINER_METADATA_URI_V4 | jq -r '.Networks[0].IPv4Addresses[0]')
 
-%{ if tls }
+%{ if tls ~}
 echo "$CONSUL_CACERT" > /tmp/consul-ca-cert.pem
-%{ endif }
-
-exec consul agent \
-  -advertise "$ECS_IPV4" \
-  -datacenter "$CONSUL_DATACENTER" \
-  -data-dir /consul/data \
-  -client 0.0.0.0 \
-%{ if gossip_encryption_enabled ~}
-  -encrypt "$CONSUL_GOSSIP_ENCRYPTION_KEY" \
 %{ endif ~}
-  -hcl 'addresses = { dns = "127.0.0.1" }' \
-  -hcl 'addresses = { grpc = "127.0.0.1" }' \
-  -hcl 'addresses = { http = "127.0.0.1" }' \
-%{ for j in retry_join ~}
-  -retry-join "${j}" \
-%{ endfor ~}
-  -hcl 'telemetry { disable_compat_1.9 = true }' \
-  -hcl 'leave_on_terminate = true' \
-  -hcl 'ports { grpc = 8502 }' \
-  -hcl 'advertise_reconnect_timeout = "15m"' \
-  -hcl 'enable_central_service_config = true' \
-%{ if tls ~}
-  -hcl 'ca_file = "/tmp/consul-ca-cert.pem"' \
-  -hcl 'auto_encrypt = {tls = true}' \
-  -hcl "auto_encrypt = {ip_san = [\"$ECS_IPV4\"]}" \
-  -hcl 'verify_outgoing = true' \
+
+cat << EOF > /consul/agent-defaults.hcl
+${consul_agent_defaults_hcl}
+EOF
+
+%{ if consul_agent_configuration_hcl != null ~}
+cat << EOF > /consul/agent-extra.hcl
+${consul_agent_configuration_hcl}
+EOF
 %{ endif ~}
-%{ if acls ~}
-  -hcl='acl {enabled = true, default_policy = "deny", down_policy = "async-cache"}' \
-  -hcl="acl { tokens { agent = \"$AGENT_TOKEN\" } }" \
+
+exec consul agent \
+    -data-dir /consul/data \
+    -config-file /consul/agent-defaults.hcl \
+%{ if consul_agent_configuration_hcl != null ~}
+    -config-file /consul/agent-extra.hcl
```

Comment:
I was originally trying to also support Consul CLI flags. But after some careful review, all Consul flags have an equivalent config file option, with the exception of a few that don't matter (`-dev`, `-config-dir`, `-config-file`, `-config-format`, and `-hcl` are the only flags without a config file option).

This now uses only config files to pass options to Consul. And we rely on [Consul's config merging behavior](https://www.consul.io/docs/agent/options) when passing multiple files:

> Configuration specified later will be merged into configuration specified earlier. In most cases, "merge" means that the later version will override the earlier. In some cases, such as event handlers, merging appends the handlers to the existing configuration. The exact merging behavior is specified for each option below.

Reply:
yeah no need to support cli flags imo